### ISSUE TITLE
[LakeFSOutputCommitter] Verify output branch is clean before starting

### ISSUE THREAD

**Issue by arielshaqed** [treeverse/lakeFS#4519]
When setting up to write verify that the output branch is clean (has no
diffs), fail if it is not.  It will be impossible to merge into a dirty
branch.

This should be configurable (default true).

**Comment by arielshaqed** [comment]
This was actually done :-/
